base/tps/src/org/dogtagpki/server/tps/rest/ProfileService.java

```diff
@@ -93,15 +100,45 @@ public Response findProfiles(String filter, Integer start, Integer size) {
         if (filter != null && filter.length() < MIN_FILTER_LENGTH) {
             throw new BadRequestException("Filter is too short.");
         }
+	//cfu
+        CMS.debug("ProfileService.j.findProfiles filter: " + filter);
+        CMS.debug("ProfileService.j.findProfiles principal name: " +
+			servletRequest.getUserPrincipal().getName());
+        try {
+            String userID = servletRequest.getUserPrincipal().getName();
+            IUGSubsystem userGroupManager = (IUGSubsystem) CMS.getSubsystem(CMS.SUBSYSTEM_UG);
+            IUser user = userGroupManager.getUser(userID);
+            if (user == null) {
+                throw new UserNotFoundException(userID);
+            }
+            CMS.debug("ProfileService.j.findProfiles getting list of authorized profiles");
+            List<String> authorizedProfiles = user.getTpsProfiles();
 
-        start = start == null ? 0 : start;
-        size = size == null ? DEFAULT_SIZE : size;
+            start = start == null ? 0 : start;
+            size = size == null ? DEFAULT_SIZE : size;
 
-        try {
             TPSSubsystem subsystem = (TPSSubsystem) CMS.getSubsystem(TPSSubsystem.ID);
             ProfileDatabase database = subsystem.getProfileDatabase();
 
-            Iterator<ProfileRecord> profiles = database.findRecords(filter).iterator();
+            Iterator<ProfileRecord> profiles = null;
+	    Collection<ProfileRecord> profilesCol = new ArrayList<ProfileRecord>();
+	    if (authorizedProfiles != null) {
+                for (String profile: authorizedProfiles) {
+                    if (profile.equals(UserResource.ALL_PROFILES)) {
+                        CMS.debug("ProfileService.j.findProfiles profile a: " + profile);
+                        profiles = database.findRecords(filter).iterator();
```

Comment:
IIUC this means that if `authorizedProfiles` contains `ALL_PROFILES` then any other profiles in `authorizedProfiles` do not matter anymore. Is that the intention?

Reply:
yes

Reply:
In that case I'd suggest separating those cases like this for clarity:
```
Collection<ProfileRecord> profiles = new ArrayList<>();

if (authorizedProfiles != null) {

    Collection<ProfileRecord> filteredProfiles = database.findRecords(filter);

    if (authorizedProfiles.contains(UserResource.ALL_PROFILES)) {
        CMS.debug("ProfileService: User allowed to access all profiles");
        profiles.addAll(filteredProfiles);

    } else {
        for (ProfileRecord profile : filteredProfiles) {
            for (authorizedProfiles.contains(profile.getID()) {
                CMS.debug("ProfileService: User allowed to access profile " + profile.getID());
                profiles.add(profile);
            }
        }
    }
}

Iterator<ProfileRecord> profileIterator = profiles.iterator();
```